package db.model;

import java.sql.SQLException;
import javax.naming.NamingException;

public class MemberDAO extends ModelBase {
	
	private static MemberDAO instance;
	
	private MemberDAO(){
		this.tableName = "member";
	}
	
	public static MemberDAO getInstance(){
		if(instance==null)
			instance=new MemberDAO();
		return instance;
	}
	
	public StringBuffer currentSql() {
		StringBuffer sql = new StringBuffer();
		sql.append("insert into ");
		sql.append(tableName);
		sql.append("values(?, ?, ?)");
		
		return sql; 
	}
	
	public void insert(MemberBean member) throws SQLException {
		super.
		Connection conn = null;
		PreparedStatement pstmt = null;
		
		try {
			conn = DBConnection.getConnection();
			conn.setAutoCommit(false);
			StringBuffer sql = currentSql();
			
			
			pstmt = conn.prepareStatement(sql.toString());
			pstmt.setString(1, member.getMem_name());
			pstmt.setString(2, member.getMem_id());
			pstmt.setString(3, member.getMem_pwd());
			
			pstmt.executeUpdate();
			conn.commit(); 
			
		} catch (ClassNotFoundException | NamingException | SQLException sqle) {
			
			conn.rollback();
			throw new RuntimeException(sqle.getMessage());
		} finally {
			try{
				if ( pstmt != null ){ pstmt.close(); pstmt=null; }
				if ( conn != null ){ conn.close(); conn=null;	}
			}catch(Exception e){
				throw new RuntimeException(e.getMessage());
			}
		} 
	} 

}
